source/plugin/advanced/message-sinks.rst

```diff
@@ -0,0 +1,114 @@
+=============
+Message Sinks
+=============
+
+A ``MessageSink`` enables you to easily control the recipients of a message. The ``org.spongepowered.api.text.sink.MessageSinks`` utility class provides methods for targeting
+appropriate recipients, as described below.
+
+Selecting MessageSink Recipients
+================================
+
+Permission-based Selection
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``MessageSinks`` can be employed to target any player that holds the specified permission.
+
+**Example: Permission-based Targeting**
+
+For example, the code below will define a ``MessageSink`` targeting all players with the `com.example.myplugin` permission
+
+.. code-block:: java
+
+    import org.spongepowered.api.text.sink.MessageSink;
+    import org.spongepowered.api.text.sink.MessageSinks;
+
+    MessageSink permissionSink = MessageSinks.toPermission("com.example.myplugin");
+
+Broadcasting
+~~~~~~~~~~~~
+
+In a broadcast, every ``CommandSource`` connected to the server will be sent the same message. This can be achieved with a ``MessageSink``, as shown below:
+
+**Example: Broadcasting**
+
+.. code-block:: java
+
+    import org.spongepowered.api.text.sink.MessageSink;
+    import org.spongepowered.api.text.sink.MessageSinks;
+
+    MessageSink broadcastSink = MessageSinks.toAll();
+
+Alternatively, you could retrieve the broadcast ``MessageSink`` provided by Sponge:
+
+.. code-block:: java
+
+    MessageSink broadcastSink = server.getBroadcastSink();
+
+Combining Message Sinks
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``MessageSinkFactory`` can also be used to combine the recipients of any arbitrary number of other ``MessageSinks``:
```

Comment:
This may be easily misunderstandable. `MessageSinks` is the utility class. Also this is the first time `MessageSinkFactory` is mentioned.
Better `MessageSinkFactory` -> `MessageSinks` class    and  `MessageSinks` - >`MessageSink`s